### ISSUE TITLE
Support publishing using API-key in Gradle plugin

### ISSUE THREAD

**Issue by skjolber** [jfrog/build-info#221]
Currently , it seems only username and password can be used for publishing:

https://github.com/jfrog/build-info/blob/master/build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/task/DeployTask.java#L128

This effectively omits users registred using SSO.

**Comment by eyalbe4** [comment]
@skjolber,
Thank you for your question.
You can use an *API Key* today by providing it as the password task input. You do need to provide the API Key (similarly to the password) with the correct username. We prefer not to directly support providing an API Key without a username. Although it can work, it is not right way to authenticate with Artifactory.
We do however intend to add support for *Access Tokens*.

**Comment by skjolber** [comment]
@eyalbe4 thanks for the quick response. Is this documented anywhere?

This issue is then reduced to the following: The gradle plugin configuration should accept an api-key property and pass it along as a password.

**Comment by eyalbe4** [comment]
Thanks @skjolber.
Added this to the [documentation](https://www.jfrog.com/confluence/display/RTF/Gradle+Artifactory+Plugin).

**Comment by skjolber** [comment]
@eyalbe4 please add it to the Gradle extension too.

**Comment by eyalbe4** [comment]
@skjolber - 
Are you referring to Gradle Artifactory Plugin documentation? That's were we added this.

**Comment by skjolber** [comment]
@eyalbe4 no, please update the configuration of the plugin, so I can write username+apiKey instead of username+password.

**Comment by yanickj** [comment]
> We do however intend to add support for Access Tokens.
Any word on if or when this will be/was added?  I can't seem to find anything in the docs for providing an access token as part of the credentials for publishing or resolving artifacts.

**Comment by yanickj** [comment]
So, after some tinkering I found that I was able to use the access __token__ in the credentials.  This seems to work with the following:

1. Decode the access token 
2. username is in the "sub" claim (everything following the final "/")
3. password is the entire token

Perhaps this is covered above (or somewhere else), but I was thrown off by API Key vs Access Token.  Also note that the username in the claim is different than the username I would use with my actual password, and it does not work with my password.   I didn't see any of this in the docs, perhaps it should be added or made more clear.  Because it appears to be undocumented I am anxious that this "fix" might not with future updates.  Would anyone from the jfrog team to be able to confirm whether or not this is the correct approach?
